fs-storage/src/folder_storage.rs

```diff
@@ -0,0 +1,598 @@
+    }
+
+    /// Load mapping from folder storage
+    fn load_fs_data(&mut self) -> Result<FolderStorageData<K, V>> {
+        if !self.path.exists() {
+            return Err(ArklibError::Storage(
+                self.label.clone(),
+                "Folder does not exist".to_owned(),
+            ));
+        }
+
+        if !self.path.is_dir() {
+            return Err(ArklibError::Storage(
+                self.label.clone(),
+                "Path is not a directory".to_owned(),
+            ));
+        }
+
+        let mut data = FolderStorageData {
+            entries: BTreeMap::new(),
+        };
+
+        self.disk_timestamps.clear();
+        self.ram_timestamps.clear();
+
+        for entry in fs::read_dir(&self.path)? {
+            let entry = entry?;
+            let path = entry.path();
+            if path.is_file()
+                && path.extension().map_or(false, |ext| ext == "bin")
+            {
+                let file_stem = path.file_stem().ok_or_else(|| {
+                    ArklibError::Storage(
+                        self.label.clone(),
+                        "Failed to extract file stem from filename".to_owned(),
+                    )
+                });
+
+                let key = file_stem?
+                    .to_str()
+                    .ok_or_else(|| {
+                        ArklibError::Storage(
+                            self.label.clone(),
+                            "Failed to convert file stem to string".to_owned(),
+                        )
+                    })?
+                    .parse::<K>()
+                    .map_err(|_| {
+                        ArklibError::Storage(
+                            self.label.clone(),
+                            "Failed to parse key from filename".to_owned(),
+                        )
+                    })?;
+
+                let mut file = File::open(&path)?;
+                let mut buffer = Vec::new();
+                file.read_to_end(&mut buffer)?;
+
+                let value: V = bincode::deserialize(&buffer).map_err(|e| {
+                    ArklibError::Storage(
+                        self.label.clone(),
+                        format!("Failed to deserialize value: {}", e),
+                    )
+                })?;
+                data.entries.insert(key.clone(), value);
+
+                if let Ok(metadata) = fs::metadata(&path) {
+                    if let Ok(modified) = metadata.modified() {
+                        self.disk_timestamps.insert(key.clone(), modified);
+                        self.ram_timestamps.insert(key, modified);
+                    }
+                }
+            }
+        }
+        Ok(data)
+    }
+
+    fn remove_files_not_in_ram(&mut self) -> Result<()> {
+        let dir = fs::read_dir(&self.path).map_err(|e| {
+            ArklibError::Storage(
+                self.label.clone(),
+                format!("Failed to read directory: {}", e),
+            )
+        })?;
```

Comment:
We should avoid re-scanning the storage directory if possible.

The only use-case for re-scanning seems to be disk corruption, or user messing with it manually. Let's do something simpler for the first version and create the follow-up task for future work on protecting the storage from user mistake or disk corruption.